src/domain/states/objects/fileTree/__tests__/SchemaTree.test.ts

```diff
@@ -62,4 +63,56 @@ describe('SchemaTree', () => {
         expect(tree.children[2].basename).toBe('aFile');
         expect(tree.children[3].basename).toBe('bFile');
     });
+
+    it('returns a flat file list', () => {
+        let fileA = new SchemaFile('File A', 'fileA', 'fileA.schema.json', 'fileA.json', []);
+        let fileB = new SchemaFile('File B', 'fileB', 'fileB.schema.json', 'fileB.json', []);
+        let fileC = new SchemaFile('File C', 'fileC', 'fileC.schema.json', 'fileC.json', []);
+        let fileD = new SchemaFile('File D', 'fileD', 'fileD.schema.json', 'fileD.json', []);
+
+        let tree = new SchemaTree([
+            new SchemaDir('Dir 1', 'dir1', [
+                fileA,
+                fileB,
+            ]),
+            new SchemaDir('Dir 2', 'dir2', [
+                fileC,
+            ]),
+            fileD,
+        ]);
+
+        let expectedFlatTree = [fileA, fileB, fileC, fileD];
+
+        expect([...new Set(tree.getFilesFlat())]).toEqual(expectedFlatTree);
```

Comment:
why do you use a Set here? I think you should test without a Set. If getFlatFiles() returns one file multiple times, then it's wrong

Reply:
I see the problem. SchemaTree.reduceFiles should look like this:
```
if (child instanceof SchemaFile) {
    return [...value, child];
} else {
    return (child as SchemaDir).children.reduce(this.reduceFiles, value);
}
```

And please update the test to don't use the Set:
```
expect(tree.getFilesFlat()).toEqual(expectedFlatTree);
```

Reply:
done